### PR TITLE
feat: added modal component

### PR DESCRIPTION
**Pull request by eakansh125**

### CODE REVIEW

File: style.css [hunk 1]
```diff
@@ -779,3 +779,41 @@ main{
     cursor: pointer;
     padding: 0.5rem;
```
Comment (yaswanthmyneni) [inline style.css:780]: changes: use css variables for colors.

**Review by yaswanthmyneni — APPROVED** [review]
You can merge after changing all colors to css variables

File: modal.html [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                <h1>Modal</h1>
+                <p>A modal is a call to action pop up which appears usually on top of the current content of the screen and
+                    freezes the contents until dismissed.
+                </p>
+                <div class="component-container">
+                    <div class="modal-container">
+                        <button id="open-modal" class="btn-small btn-outlined">OPEN MODAL</button>
+                        <div id="myModal" class="modal">
+                            <div class="modal-content font-bold">
+                                <p>Congratulatins, you are a modal opener!</p>
```
Comment (parvez-rashid) [inline modal.html]: Spelling error.

File: style.css (hunk 1 above)
Comment (parvez-rashid) [inline style.css:780]: You can use px here, generally scaling such a small value with rem is not desirable.

File: modal.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+
+var openModalbtn = document.querySelector("#open-modal")
+
+var modal = document.getElementById("myModal");
+
+var span = document.querySelector("#close-modal");
```
Comment (kaustubh-jsr) [inline modal.js]: Try using ES6+ syntax instead of var.

**Review by parvez-rashid — APPROVED** [review]
UI Review: Modal component is spot-on.

Code-wise, I've a couple of comments, have a look. Other than that, looks well-structured and neat. Can be merged.

File: modal.js (hunk 3 above)
Comment (kaustubh-jsr) [reply modal.js]: Although this doesn't give error, but you can try renaming span to something more specific, like 'closeModalBtn'.

File: modal.html (hunk 2 above)
Comment (eakansh125) [reply modal.html]: Haha, will correct it :D

File: style.css (hunk 1 above)
Comment (eakansh125) [reply style.css:780]: okay, will keep that in mind!

File: modal.js (hunk 3 above)
Comment (kaustubh-jsr) [reply modal.js]: The UI looks awesome, but this might be an oversight but there are no code snippets visible in the deploy preview.

Comment (eakansh125) [reply modal.js]: sure, will change that, thanks!

**Pull request merged**